app/src/main/java/com/sopt/gongbaek/presentation/ui/main/MainNavigator.kt

```diff
@@ -0,0 +1,74 @@
+package com.sopt.gongbaek.presentation.ui.main
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.navigation.NavDestination
+import androidx.navigation.NavGraph.Companion.findStartDestination
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.currentBackStackEntryAsState
+import androidx.navigation.compose.rememberNavController
+import androidx.navigation.navOptions
+import com.sopt.gongbaek.presentation.model.NavigationRoute
+import com.sopt.gongbaek.presentation.type.MainBottomNavBarTabType
+import com.sopt.gongbaek.presentation.ui.grouplist.navigation.navigateGroupListNavGraph
+import com.sopt.gongbaek.presentation.ui.home.navigation.navigateHomeNavGraph
+import com.sopt.gongbaek.presentation.ui.mygroup.navigation.navigateMyGroupNavGraph
+
+class MainNavigator(
+    val navController: NavHostController
+) {
+    val startDestination = NavigationRoute.MainBottomNavBarTabRoute.HOME_TAB
+
+    private val currentDestination: NavDestination?
+        @Composable get() = navController.currentBackStackEntryAsState().value?.destination
+
+    val currentMainBottomNavBarTab: MainBottomNavBarTabType?
+        @Composable get() = currentDestination
+            ?.route
+            ?.let(MainBottomNavBarTabType.Companion::find)
+
+    fun navigate(mainBottomNavBarTabType: MainBottomNavBarTabType) {
+        val navOptions = navOptions {
+            popUpTo(navController.graph.findStartDestination().id) {
+                saveState = true
+            }
+            launchSingleTop = true
+            restoreState = true
+        }
+        when (mainBottomNavBarTabType) {
+            MainBottomNavBarTabType.GROUP_LIST -> navController.navigateGroupListNavGraph(navOptions)
+
+            MainBottomNavBarTabType.MY_GROUP -> navController.navigateMyGroupNavGraph(navOptions)
+
+            MainBottomNavBarTabType.HOME -> navController.navigateHomeNavGraph(navOptions)
+
+            MainBottomNavBarTabType.TIMETABLE -> {}
+
+            MainBottomNavBarTabType.MY_PAGE -> {}
+        }
+    }
+
+    private fun popBackStack() {
+        navController.popBackStack()
+    }
```

Comment:
[3]: 단일 표현식 사용은 어떤지 의견을 남겨봅니다!
```suggestion
    private fun popBackStack() = navController.popBackStack()
```

Reply:
좋습니다! 반영하겠습니다!